ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -697,7 +950,297 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
+        downloadFailed(forKeyboards: [], error: error)
+        return
+      }
+      //put the data at a local  file  URL
+      var destinationUrl = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
+      destinationUrl.appendPathComponent("\(lexicalModelKMPURL.lastPathComponent).zip")
+      do {
+        //write the URL contents to a local file URL
+        try data.write(to: destinationUrl)
+        installLexicalModelPackage(downloadedPackageFile: destinationUrl)
+        log.info("lexical model download complete")
+      } catch {
+        log.error("writing or opening downloaded lexical model KMP failed")
+      }
+    } else {
+      log.info("\(lexicalModelPackageURLString) is not a URL string?")
+      // might want to download the .js file directly, then, instead
+    }
+  }
+  
+  /// Starts the process of fetching the package file of the lexical model for the given language ID
+  ///   first it fetches the list of lexical models for the given language
+  ///   then it takes the first of the list and download the KMP package file and asks the app to open it (like adhoc download)
+  /// - Parameters:
+  ///   - languageID: the bcp47 string of the desired language
+  public func downloadLexicalModelsForLanguageIfExists(languageID: String) {
+    //get list of lexical models for this languageID  /?q=bcp47:en
+    func listCompletionHandler(lexicalModels: [LexicalModel]?, error: Error?) -> Void {
+      if let error = error {
+        log.info("Failed to fetch lexical model list for "+languageID+". error: "+(error as! String))
+        self.downloadFailed(forLanguageID: languageID, error: error) //???forKeyboards
+      } else {
+        log.info("Fetched lexical model list for "+languageID+".")
+        // choose which of the lexical models to download
+        //  for now, this just downloads the first one
+        let chosenIndex = 0
+        if let lexicalModel = lexicalModels?[chosenIndex] {
+          downloadLexicalModelPackage(string: lexicalModel.packageFilename)
+        } else {
+          log.info("no error, but no lexical model in list, either!")
+        }
+      }
+    }
+    
+    apiLexicalModelRepository.fetchList(languageID: languageID, completionHandler: listCompletionHandler)
+  }
+
+  /// Asynchronously fetches the .js file for the lexical model with given IDs.
+  /// See `Notifications` for notification on success/failiure.
+  /// - Parameters:
+  ///   - isUpdate: Keep the lexical model files on failure
+  ///   - fetchRepositoryIfNeeded: Fetch the list of lexical models from the API if necessary.
+  public func downloadLexicalModel(withID lexicalModelID: String,
+                                   languageID: String,
+                                   isUpdate: Bool,
+                                   fetchRepositoryIfNeeded: Bool = true) {
+//    guard let lexicalModels = apiLexicalModelRepository.lexicalModels,
```

Comment:
So... is there a reason we're leaving this non-functional method in?  Is implementation planned in the future?

We should still have a reasonable reference from the equivalent version for keyboards, so I'd vote we remove it.

Reply:
I'd be happy removing it.  Something like it I think would be used if lexical models were pickable,  say from the list of available ones for your language. It's analogous from picking a keyboard to download, isn't it?

Reply:
We will need a picker for lexical models for a language soon; see https://docs.google.com/document/d/1LQW6F1GpqmMxOXm-tEvranUu_T6ck_M0Llk2fyJmqcI/edit#bookmark=id.wdyzeog744g4